tests/unit/backend/layers/persistence/persistence_mock.py

```diff
@@ -0,0 +1,176 @@
+
+    def delete_collection_version(self, version_id: str) -> None:
+        # Can only delete an unpublished collection
+        del self.collections_versions[version_id]
+
+    def get_collection_version(self, version_id: str) -> CollectionVersion:
+        return self.collections_versions[version_id]
+
+    # MAYBE
+    def publish_collection_version(self, collection_id: str, version_id: str) -> None:
+        self.collections[collection_id] = version_id
+        self.collections_versions[version_id].published_at = datetime.utcnow()
+
+    # OR
+    def update_collection_version_mapping(self, collection_id: str, version_id: str) -> None:
+        self.collections[collection_id] = version_id
+
+    def set_collection_version_published_at(self, version_id) -> None:
+        self.collections_versions[version_id].published_at = datetime.utcnow()
+
+    # END OR
+
+    def get_dataset(self, dataset_id: str) -> DatasetVersion:
+        version_id = self.datasets[dataset_id]
+        return self.datasets_versions[version_id]
+
+    def get_dataset_version(self, version_id: str) -> DatasetVersion:
+        return self.datasets_versions[version_id]
+
+    def get_all_datasets(self) -> Iterable[DatasetVersion]:  # TODO: add filters if needed
+        for version_id, dataset_version in self.datasets_versions.items():
+            if version_id in self.datasets.values():
+                yield dataset_version
+
+    def get_dataset_artifacts(self, dataset_id: str) -> List[DatasetArtifact]:
+        dataset = self.get_dataset(dataset_id)
+        return dataset.artifacts
+
+    def create_dataset(self, collection_version_id: str, dataset_metadata: DatasetMetadata) -> None:
+        # Creates a dataset and initializes it with one version
+        dataset_id = self._id()
+        version_id = self._id()
+        version = DatasetVersion(
+            id=dataset_id,
+            version_id=version_id,
+            processing_status=None,
+            metadata=dataset_metadata,
+            artifacts=[]
+        )
+        self.datasets_versions[version_id] = version
+        self.datasets[dataset_id] = version_id
+        # Register the dataset to the original collection
+        self.collections_versions[collection_version_id].datasets.append(version)
+
+    def create_dataset_version(self, dataset_id: str, dataset_metadata: DatasetMetadata) -> str:
+        # Unused for now
+        pass
+
+    def create_dataset_artifact(self, version_id: str, artifact: DatasetArtifact) -> None:
+        version = self.datasets_versions[version_id]
+        version.artifacts.append(artifact)
+
+    def update_dataset_processing_status(self, version_id: str, status: DatasetStatus) -> None:
+        dataset_version = self.datasets_versions[version_id]
+        dataset_version.processing_status = status
+
+    def add_dataset_to_collection_version(self, version_id: str, dataset_id: str) -> None:
+        # Not needed for now - create_dataset does this
```

Comment:
Also, why would a _canonical_ dataset need to be mapped to a collection version? We have `add_dataset_to_collection_version_mapping(self, collection_version_id: str, dataset_version_id: str)` which is associating a dataset _version_ to a collection _version_. Isn't that all we'll need?

Reply:
moving this comment to the interface...

Reply:
in fact, this is not even an interface method. :)  maybe just remove it here?

Reply:
Yep, this is now outdated. The mock hasn't kept up with the interface, since we're making a lot of changes, I will update it when we're settled on the interface.